Description: Load the JNI library directly instead of using the NAR system
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/junixsocket-common/src/main/java/org/newsclub/net/unix/NativeUnixSocket.java
+++ b/junixsocket-common/src/main/java/org/newsclub/net/unix/NativeUnixSocket.java
@@ -32,12 +32,7 @@
 
   static {
     try {
-      Class.forName("org.newsclub.net.unix.NarSystem").getMethod("loadLibrary").invoke(null);
-    } catch (ClassNotFoundException e) {
-      throw new IllegalStateException(
-          "Could not find NarSystem class.\n\n*** ECLIPSE USERS ***\nIf you're running from "
-              + "within Eclipse, please try closing the \"junixsocket-native-common\" "
-              + "project\n", e);
+      System.loadLibrary("junixsocket.system");
     } catch (Exception e) {
       throw new IllegalStateException(e);
     }
